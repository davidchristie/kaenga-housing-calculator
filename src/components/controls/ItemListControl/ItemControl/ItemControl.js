import React, { Component, PropTypes } from 'react'
import { Button, ListGroupItem } from 'react-bootstrap'

class DefaultItemComponent extends Component {
  render () {
    return <div />
  }
}

class ItemControl extends Component {
  render () {
    const {
      index,
      item,
      itemComponent: ItemComponent,
      remove
    } = this.props
    return (
      <ListGroupItem>
        <h4>#{index + 1}</h4>
        <ItemComponent item={item} />
        <Button onClick={remove}>Remove</Button>
      </ListGroupItem>
    )
  }
}

ItemControl.defaultProps = {
  item: '',
  itemComponent: DefaultItemComponent,
  index: 0
}

ItemControl.propTypes = {
  item: PropTypes.string.isRequired,
  itemComponent: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  remove: PropTypes.func
}

export default ItemControl
