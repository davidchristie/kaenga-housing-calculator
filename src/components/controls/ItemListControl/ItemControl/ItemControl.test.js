import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import ItemControl from './ItemControl'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<ItemControl />)).toMatchSnapshot()
})
