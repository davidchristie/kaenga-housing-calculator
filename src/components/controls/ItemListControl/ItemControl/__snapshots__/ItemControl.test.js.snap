exports[`test matches snapshot 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  listItem={false}>
  <h4>
    #
    1
  </h4>
  <DefaultItemComponent
    item="" />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={undefined}>
    Remove
  </Button>
</ListGroupItem>
`;
